workspaces/marketplace/plugins/marketplace/src/components/InstalledPackages/InstalledPackagesTable.tsx

```diff
@@ -0,0 +1,365 @@
+                  <Box component="span" display="inline-flex">
+                    <IconButton
+                      size="small"
+                      disabled
+                      sx={{ color: theme => theme.palette.action.disabled }}
+                    >
+                      <DeleteIcon />
+                    </IconButton>
+                  </Box>
+                </Tooltip>
+              ) : (
+                <IconButton
+                  size="small"
+                  sx={{ color: theme => theme.palette.text.primary }}
+                  onClick={() => {
+                    // TODO: Implement delete functionality
+                  }}
+                >
+                  <DeleteIcon />
+                </IconButton>
+              )}
+              {disabled ? (
+                <Tooltip title={tooltipTitle}>
+                  <Box component="span" display="inline-flex">
+                    <IconButton
+                      size="small"
+                      disabled
+                      sx={{ color: theme => theme.palette.action.disabled }}
+                    >
+                      <FileDownloadOutlinedIcon />
+                    </IconButton>
+                  </Box>
+                </Tooltip>
+              ) : (
+                <IconButton
+                  size="small"
+                  sx={{ color: theme => theme.palette.text.primary }}
+                  onClick={() => {
+                    // TODO: Implement download functionality
+                  }}
+                >
+                  <FileDownloadOutlinedIcon />
+                </IconButton>
+              )}
+            </Box>
+          );
+        },
+        sorting: false,
+      },
+    ],
+    [location.pathname, location.search],
+  );
+  const fetchData = async (
+    query: Query<InstalledPackageRow>,
+  ): Promise<QueryResult<InstalledPackageRow>> => {
+    const { page = 0, pageSize = 5 } = query || {};
+
+    try {
+      // Ensure data available; avoid re-fetching on search or pagination
+      const installed = installedQuery.data ?? [];
+      const packagesResponse = packagesQuery.data ?? { items: [] as any[] };
+
+      const entitiesByName = new Map(
+        (packagesResponse.items ?? []).map(entity => [
+          (entity.metadata?.name ?? '').toLowerCase(),
+          entity,
+        ]),
+      );
+
+      // Build rows for ALL installed items; if no matching entity, mark missing
+      const rows: InstalledPackageRow[] = installed.map(p => {
+        const normalized = p.name
+          .replace(/[@/]/g, '-')
+          .replace(/-dynamic$/, '')
+          .replace(/^-+/, '')
+          .toLowerCase();
+        const entity = entitiesByName.get(normalized) as any | undefined;
+        const rawName = entity
+          ? (entity.metadata?.title as string) ||
+            (entity.metadata?.name as string)
+          : getReadableName(p.name);
+        const cleanedName = rawName.replace(/\s+(frontend|backend)$/i, '');
```

Comment:
I know this is copied but this looks like a lot of code to me that the backstage table is already doing? But maybe I'm wrong in the detail.

Fine for now.